### DISCUSSION TITLE
Add a new (custom) button for global rejection of all categories of cookies (empty array)

### DISCUSSION THREAD

**Discussion by Group24Steven** [orestbida/cookieconsent#747] [Ideas]
Currently (as well in the demo playground), the option "reject all" is a hidden "accept only necessary" one... May you please add to a subversion of v3 a new button (at least make it toggleable), which is an actual "reject all" one, so you have an entire set of options being:

- Accept all (as already recently possible)
- Reject all
- Accept only necessary (as already recently possible, being hidden as "reject all" inside all v3 demos online)
- Choose from categories (as already recently possible)

This would also have the benefit to prevent conflicts with people who might argue about the GDPR compliance.
Besides that a custom button which can be written in sort of a plugin and used with vanilla-cookieconsent might be a workaround for this?

**Comment by proudyy** [comment]
**Edit:** GDPR compliance is still given with such functionality, however (as already described) this might lead to increased requests about questioning it. Furthermore, a "reject all" option would allow an even more separation of the cookies.

**For example:**
- Accept all | Accepts all cookies (All below and things like marketing or advertisement data, hardware/device data, etc.)
- Accept only necessary | Necessary ones (including the preference cookie and other technical ones such as general device data etc.)
- Reject all | Closes the dialog (preferences one or the compact one being in either left or right corner).

The "reject all" option then would of course not store a preference cookie. Devs would then have to just decline all technical stuff which explicitly relies on cookies and notifies the user to accept such cookies.

Let me know your thoughts of this, thanks in advance! 😊
